web/modules/custom/parser/src/Writer/DB/DiscountWriter.php

```diff
@@ -47,21 +38,14 @@ public function write(array $rawdata, $truncate, DrupalStyle $io) {
    */
   private function mapdata(array $rawdata) {
     $headers = $this->discountHeaders();
-    $discounts = [];
-
-    foreach ($rawdata as $key => $data) {
-      preg_match('/[ADFJMNOS][aceopu][bcglnprtvy]-\d{2}-\d{4}/', $this->file[$key], $phpDate);
-      array_shift($data);
-
-      foreach ($data as $row) {
-        $row[] = strtotime($phpDate[0]);
+    $discount = [];
+    $timestamp = strtotime($this->date);
 
-        if ($row[0] != NULL) {
-          array_push($discounts, array_combine($headers, $row));
-        }
-      }
+    foreach ($rawdata as $row) {
+      array_push($row, $timestamp);
```

Comment:
If you use array_push() to add one element to the array, it's better to use $array[] = because in that way there is no overhead of calling a function. 
From: http://php.net/manual/en/function.array-push.php